framework/src/base/FEProblemBase.C

```diff
@@ -3132,9 +3130,17 @@ FEProblemBase::setCoupling(Moose::CouplingType type)
 void
 FEProblemBase::setCouplingMatrix(CouplingMatrix * cm)
 {
+  // TODO: Deprecate method
+
+  _coupling = Moose::COUPLING_CUSTOM;
+  _cm.reset(cm);
+}
+
+void
+FEProblemBase::setCouplingMatrix(std::unique_ptr<CouplingMatrix> cm)
+{
   _coupling = Moose::COUPLING_CUSTOM;
-  delete _cm;
-  _cm = cm;
+  _cm = std::move(cm);
```

Comment:
I think `setCouplingMatrix` should take its argument by r-value reference, right?  See also, the discussion in this stack overflow [article](http://stackoverflow.com/questions/8114276/how-do-i-pass-a-unique-ptr-argument-to-a-constructor-or-a-function).